src/main/java/org/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -96,329 +70,466 @@ public ParserResult importDatabase(BufferedReader input) throws IOException {
+        };
     }
 
-    private void putPublisherOrEdition(Map<Field, String> fields, String elementName,
-                                       StringPlusLanguagePlusSupplied pubOrEd) {
-        if ("publisher".equals(elementName)) {
-            putIfValueNotNull(fields, StandardField.PUBLISHER, pubOrEd.getValue());
-        } else if ("edition".equals(elementName)) {
-            putIfValueNotNull(fields, StandardField.EDITION, pubOrEd.getValue());
+    private void parseIdentifier(Map<Field, String> fields, Identifier identifier, BibEntry entry) {
+        String type = identifier.type();
+        if ("citekey".equals(type) && entry.getCitationKey().isEmpty()) {
+            entry.setCitationKey(identifier.value());
+        } else if (!"local".equals(type) && !"citekey".equals(type)) {
+            // put all identifiers (doi, issn, isbn,...) except of local and citekey
+            putIfValueNotNull(fields, FieldFactory.parseField(identifier.type()), identifier.value());
         }
     }
 
-    private void putDate(Map<Field, String> fields, String elementName, DateDefinition date) {
-        if (date.getValue() != null) {
+    private void putDate(Map<Field, String> fields, String elementName, String date) {
+        if (date != null) {
             switch (elementName) {
-                case "dateIssued":
+                case "dateIssued" -> {
                     // The first 4 digits of dateIssued should be the year
-                    fields.put(StandardField.YEAR, date.getValue().replaceAll("[^0-9]*", "").replaceAll("\\(\\d?\\d?\\d?\\d?.*\\)", "\1"));
-                    break;
-                case "dateCreated":
+                    fields.put(StandardField.YEAR, date.replaceAll("[^0-9]*", "").replaceAll("\\(\\d?\\d?\\d?\\d?.*\\)", "\1"));
+                }
+                case "dateCreated" -> {
                     // If there was no year in date issued, then take the year from date created
-                    fields.computeIfAbsent(StandardField.YEAR, k -> date.getValue().substring(0, 4));
-                    fields.put(new UnknownField("created"), date.getValue());
-                    break;
-                case "dateCaptured":
-                    fields.put(new UnknownField("captured"), date.getValue());
-                    break;
-                case "dateModified":
-                    fields.put(new UnknownField("modified"), date.getValue());
-                    break;
-                default:
-                    break;
+                    fields.computeIfAbsent(StandardField.YEAR, k -> date.substring(0, 4));
+                    fields.put(new UnknownField("created"), date);
+                }
+                case "dateCaptured" -> {
+                    fields.put(new UnknownField("captured"), date);
+                }
+                case "dateModified" -> {
+                    fields.put(new UnknownField("modified"), date);
+                }
             }
         }
     }
```

Comment:
Instead of building a custom year parser, you could use `our model.entry.Date.parse()` method. That way, we can avoid unnecessary data loss by parsing.

Not sure about created and modified field. We use the fields `CREATIONDATE` and `MODIFICATIONDATE` for internal description of changes to bibentries. Question is: Should we import them from the source and interpret them like our internal ones, or should we forget about them @koppor @Siedlerchr ?

Reply:
Updated to use the `model.entry.Date.parse()` method

Reply:
Can you fill the date field in biblatex mode too? In biblatex mode, this is the preferred way to keep dates (you could use getNormalized). Also the month field could could be filled, if there is a month available...

Reply:
Thanks, I missed that - I'll add these fields soon

Reply:
Hi @calixtus, I've added the date and month fields. I've also modified a couple of test files since they do not include the `date` field currently

Reply:
> Not sure about created and modified field. We use the fields `CREATIONDATE` and `MODIFICATIONDATE` for internal description of changes to bibentries. Question is: Should we import them from the source and interpret them like our internal ones, or should we forget about them @koppor @Siedlerchr ?

I would opt for mapping `dateCaptured` to `CREATIONDATE` and `dateModified` to `MODIFICATIONDATE`. This now has to go into a follow-up PR.